src/rest.cpp

```diff
@@ -950,6 +1001,7 @@ static const struct {
       {"/rest/mempool/contents", rest_mempool_contents},
       {"/rest/headers/", rest_headers},
       {"/rest/getutxos", rest_getutxos},
+      {"/rest/deploymentinfo/", rest_deploymentinfo},
```

Comment:
This doesn't work with `/rest/deploymentinfo.json`

```diff
--- a/test/functional/interface_rest.py
+++ b/test/functional/interface_rest.py
@@ -380,8 +380,10 @@ class RESTTest (BitcoinTestFramework):
 
         self.log.info("Test the /deploymentinfo URI")
         json_obj = self.test_rest_request(f'/deploymentinfo/{newblockhash[0]}')
+        assert_equal(json_obj, self.test_rest_request(f'/deploymentinfo'))
         deployment_info = self.nodes[0].getdeploymentinfo(newblockhash[0])
         assert_equal(deployment_info, json_obj)
+        assert(deployment_info != self.test_rest_request(f'/deploymentinfo/{bb_hash}'))
 
         json_obj = self.test_rest_request('/deploymentinfo/')
         deployment_info = self.nodes[0].getdeploymentinfo()
```

Reply:
Just noticed that `newblockhash[0]` and `bb_hash` are the same, so the assertion: `assert(deployment_info != self.test_rest_request(f'/deploymentinfo/{bb_hash}'))` would fail.

Reply:
Fixed `/rest/deploymentinfo.json`

Reply:
`bb_hash` is set to the best block, *before* `newblockhash[0]` is mined...

Reply:
Before? 

`newblockhash = self.generate(self.nodes[1], 1)` -> L348
`bb_hash = self.nodes[0].getbestblockhash()` -> L372